venafi/resource_venafi_certificate.go

```diff
@@ -531,6 +562,33 @@ func enrollVenafiCertificate(ctx context.Context, d *schema.ResourceData, cl end
 	if origin == csrService {
 		req.CsrOrigin = certificate.ServiceGeneratedCSR
 	}
+
+	//setting DN values Org, Organization Units, Country, State, Locality(City)
+	country := d.Get("country").(string)
+	if country != "" {
+		req.Subject.Country = []string{country}
+	}
+	state := d.Get("state").(string)
+	if state != "" {
+		req.Subject.Province = []string{state}
+	}
+	locality := d.Get("locality").(string)
+	if locality != "" {
+		req.Subject.Locality = []string{locality}
+	}
+	org := d.Get("organization").(string)
+	if org != "" {
+		req.Subject.Organization = []string{org}
+	}
+	orUnitsNum := d.Get("organizational_units.#").(int)
+	if orUnitsNum > 0 {
+		for i := 0; i < orUnitsNum; i++ {
+			key := fmt.Sprintf("organizational_units.%d", i)
```

Comment:
You don't need to iterate the slice based on the index.
You can assert the type of the slice and then assert the element's type when iterating it:

```
orgUnits := d.Get("organizational_units").([]interface{})
```
Now you have a list of interfaces ready to iterate:
```
for i, item := range orgUnits{
    itemStr := item.(string)
    req.Subject.OrganizationalUnit = append(req.Subject.OrganizationalUnit, itemStr)
}
```

Reply:
done